src/main/java/gregtech/client/utils/RenderUtil.java

```diff
@@ -604,4 +605,34 @@ public static void hookDepthTexture(Framebuffer fbo, int depthTexture) {
             OpenGlHelper.glFramebufferTexture2D(OpenGlHelper.GL_FRAMEBUFFER, OpenGlHelper.GL_DEPTH_ATTACHMENT, GL11.GL_TEXTURE_2D, depthTexture, 0);
         }
     }
+
+    /**
+     * Makes given BakedQuad emissive; specifically, it makes a new UnpackedBakedQuad with
+     * constant lightmap coordination and {@code applyDiffuseLighting} set to {@code false}.
+     * The other properties, such as textures, tint color and other vertex data will be copied from
+     * the template.<p>
+     * Note that this method just returns {@code quad} if Optifine is installed.
+     *
+     * @param quad Template BakedQuad.
+     * @return New UnpackedBakedQuad with emissive property
+     */
+    public static BakedQuad makeEmissive(BakedQuad quad) {
+        if (FMLClientHandler.instance().hasOptifine()) return quad;
+        VertexFormat format = quad.getFormat();
+        if (!format.getElements().contains(DefaultVertexFormats.TEX_2S)) {
+            format = new VertexFormat(quad.getFormat());
+            format.addElement(DefaultVertexFormats.TEX_2S);
+        }
+        UnpackedBakedQuad.Builder builder = new UnpackedBakedQuad.Builder(format) {
+            @Override
+            public void put(int element, float... data) {
+                if (this.getVertexFormat().getElement(element) == DefaultVertexFormats.TEX_2S)
+                    super.put(element, 480.0f / 0xFFFF, 480.0f / 0xFFFF);
+                else super.put(element, data);
+            }
+        };
```

Comment:
Is this anonymous class needed / used for anything?

Reply:
The `put` method override is to change value for lightmap coordination (tex_2s) to fixed value. It's needed for no-bloom emissive textures as well as item models (since they can't have bloom effect)